proparse/src/main/java/org/prorefactor/core/JPNodeQuery.java

```diff
@@ -11,33 +11,60 @@
 package org.prorefactor.core;
 
 import java.util.ArrayList;
+import java.util.EnumSet;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 class JPNodeQuery implements ICallback<List<JPNode>> {
   private final List<JPNode> result = new ArrayList<>();
-  private final Set<Integer> findTypes;
+  private final Set<ABLNodeType> findTypes;
   private final boolean stateHeadOnly;
   private final boolean mainFileOnly;
 
+  /**
+   * @deprecated Since 2.1.3, use {@link JPNodeQuery#JPNodeQuery(ABLNodeType, ABLNodeType...)}
+   */
+  @Deprecated
   public JPNodeQuery(Integer... types) {
     this(false, false, types);
   }
 
+  /**
+   * @deprecated Since 2.1.3, use {@link JPNodeQuery#JPNodeQuery(boolean, ABLNodeType, ABLNodeType...)}
+   */
+  @Deprecated
   public JPNodeQuery(boolean stateHeadOnly, Integer... types) {
     this(stateHeadOnly, false, types);
   }
 
+  /**
+   * @deprecated Since 2.1.3, use {@link JPNodeQuery#JPNodeQuery(boolean, boolean, ABLNodeType, ABLNodeType...)}
+   */
+  @Deprecated
   public JPNodeQuery(boolean stateHeadOnly, boolean mainFileOnly, Integer... types) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1133)